.github/workflows/create-and-upload-coverage-report.yml

```diff
@@ -0,0 +1,94 @@
+#
+# This source file is part of the Stanford Biodesign Digital Health Group open-source organization
+#
+# SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Create and upload coverage report
+
+on:
+  workflow_call:
+    inputs:
+      coveragereports:
+        description: |
+          A string containing all names of the .xcresult files that are downloaded from the artifacts of previous jobs.
+          Multiple values are separated by a space character.
+        required: false
+        type: string
+      coveragereport_lcov:
+        description: |
+          A string with the path to an lcov file that are downloaded from the artifacts of previous jobs.
+        required: false
+        type: string
```

Comment:
We should probably also allow the input of multiple reports that we merge? Naming should then probably be `coveragereports_lcov`